yorkie/backend/sync/pubsub.go

```diff
@@ -78,6 +78,11 @@ func (s *Subscription) SubscriberID() string {
 	return s.subscriber.ID.String()
 }
 
+// UpdateMetadata updates metadata of subscriber.
+func (s *Subscription) UpdateMetadata(subscriber types.Client) {
+	s.subscriber.Metadata = subscriber.Metadata
+}
```

Comment:
Is there any need to lock the mutex in this part🤔??

Reply:
It would be nice to write test cases that modify the metadata at the same time.

Reply:
@ppeeou @hackerwins Thank you for your comment!

This is the reason why the method was not locked.
To access `Subscription`, you can access it through `PubSub.subscriptionsMap`. And since `subscriptionsMap` is private, I thought that PubSub's mutex would be sufficient. This is because all PubSub methods use the same mutex. As a result, it was determined that it is safe to call other `PubSub` methods at the same time.

At the same time, when modifying the metadata, it was confirmed that the order in which the metadata is updated and the order in which it is published does not match. The UpdateMetadata API has been improved so that can be locked by the client ID. (c92ff31)